When the system is busy it may happen that the command actually
completed but it took more than the specified timeout till the
task executing the command was actually given CPU time. This test
checks that the completion is really missing before failing.

Signed-off-by: Eli Cohen <eli@mellanox.co.il>

---

---
 drivers/net/mlx4/cmd.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

Index: ofed_kernel/drivers/net/mlx4/cmd.c
===================================================================
--- ofed_kernel.orig/drivers/net/mlx4/cmd.c
+++ ofed_kernel/drivers/net/mlx4/cmd.c
@@ -282,10 +282,11 @@ static int mlx4_cmd_wait(struct mlx4_dev
 	mlx4_cmd_post(dev, in_param, out_param ? *out_param : 0,
 		      in_modifier, op_modifier, op, context->token, 1);
 
-	if (!wait_for_completion_timeout(&context->done, msecs_to_jiffies(timeout))) {
-		err = -EBUSY;
-		goto out;
-	}
+	if (!wait_for_completion_timeout(&context->done, msecs_to_jiffies(timeout)))
+		if (!context->done.done) {
+			err = -EBUSY;
+			goto out;
+		}
 
 	err = context->result;
 	if (err)
